File: src/contracts/utils/demo2.py
import time
from algosdk.v2client import algod
from algosdk import encoding, constants, transaction
from .accounts import get_accounts
from src.contracts.escrow import EscrowAccount
from .helpers import compile_contract
from ..crowdfund_contract import FundingProject
ACCOUNT_NAMES = ['platform', 'creator', 'doner']

ALGOD_ADDRESS = "https://testnet-algorand.api.purestake.io/"
ALGOD_TOKEN = ""
HEADERS = {
    "X-API-Key": "ROs0KPlr9G71RxDhah1D2qp0sjiJQnV2NGGFOAUf",
}
print("Headers:", HEADERS)

algod_client = algod.AlgodClient(ALGOD_TOKEN, ALGOD_ADDRESS, headers=HEADERS)
CROWDFUND_CONTRACT = FundingProject()
ESCROW_ACCOUNT = EscrowAccount()

class SimpleDemoApp():
    def __init__(self):
        self.accounts = {name: acc for name, acc in zip(ACCOUNT_NAMES, get_accounts())}
        self.client = algod.AlgodClient(ALGOD_TOKEN, ALGOD_ADDRESS)
        
        # Logging the creation process
        print("Initializing SimpleDemoApp...")
        print(f"Using accounts: {self.accounts}")

        # Compile the contract
        print("Compiling Crowdfund Contract...")
        self.crowdfund_approval = compile_contract(CROWDFUND_CONTRACT.approval_program(), self.client)
        print("Crowdfund Contract compiled successfully.")
        
    def run(self):
        # This is a placeholder for your main logic
        print("Running the SimpleDemoApp...")

if __name__ == '__main__':
    print("Starting the SimpleDemoApp...")
    app = SimpleDemoApp()
    app.run()
    print("Execution completed.")

File: src/contracts/utils/demo2.js
const algosdk = require('algosdk');
// const fs = require('fs');
if (typeof window === 'undefined') {
    // Server-side only code
    const fs = require('fs');
  }

const baseServer = "https://testnet-algorand.api.purestake.io/ps2";
const token = {
    'X-API-Key': 'ROs0KPlr9G71RxDhah1D2qp0sjiJQnV2NGGFOAUf'
};

let algodClient = new algosdk.Algodv2(token, baseServer, '');

// TODO: Implement get_accounts, compile_contract, and other required functions
// These functions will probably require additional implementations based on your Python logic

class SimpleDemoApp {
    constructor() {
        this.accounts = this.getAccounts();
        this.client = new algosdk.Algodv2(token, baseServer, '');

        console.log("Initializing SimpleDemoApp...");
        console.log(`Using accounts: ${JSON.stringify(this.accounts)}`);

        // Compile the contract
        console.log("Compiling Crowdfund Contract...");
        this.crowdfundApproval = this.compileContract("YOUR_APPROVAL_TEAL_CODE_HERE");
        console.log("Crowdfund Contract compiled successfully.");
    }

    getAccounts() {
        // TODO: Implement this function based on the Python get_accounts function
    }

    compileContract(tealCode) {
        // TODO: Implement this function based on the Python compile_contract function
    }

    run() {
        console.log("Running the SimpleDemoApp...");
    }
}

console.log("Starting the SimpleDemoApp...");
let app = new SimpleDemoApp();
app.run();
console.log("Execution completed.");
